src/BookStore/Infrastructure/ApiPlatform/State/Processor/UpdateBookProcessor.php

```diff
@@ -33,9 +33,10 @@ public function __construct(
     public function process($data, Operation $operation, array $uriVariables = [], array $context = []): mixed
     {
         Assert::isInstanceOf($data, BookResource::class);
+        Assert::isInstanceOf($context['previous_data'], BookResource::class);
```

Comment:
```suggestion
        Assert::isInstanceOf($context['previous_data'] ?? null, BookResource::class);
```

Reply:
Previous data should always be here, is it necessary?

Reply:
Indeed, I think we can trust API Platform to not introduce BC breaks :wink: